ipn/ipnlocal/local.go

```diff
@@ -5442,7 +5434,9 @@ func (b *LocalBackend) enterStateLockedOnEntry(newState ipn.State, unlock unlock
 	activeLogin := b.activeLogin
 	authURL := b.authURL
 	if newState == ipn.Running {
-		b.resetAuthURLLocked()
+		if oldState != ipn.Running {
+			b.resetAuthURLLocked()
+		}
```

Comment:
I think it's debatable if we need this here at all, as we also call this in `SetControlClientStatus` when we see a "LoginFinished" controlclient status? (And all tests pass if you delete this line)

But this is the fix for the 95% case race condition, where "entering" the running state while already running would clear it.

Reply:
Comment added